contracts/stellar-interchain-token-service/src/contract.rs

```diff
@@ -311,6 +311,17 @@ impl InterchainTokenServiceInterface for InterchainTokenService {
 
         Ok(())
     }
+
+    #[only_owner]
+    #[allow_during_migration]
+    fn migrate_token(
```

Comment:
Another interesting pattern to keep in mind, is that we could dynamically call this method during interchain_transfer to auto-upgrade any token_id that's not yet upgraded. We used this approach for EVM ITS upgrade where there's a lot of tokens already deployed, and the upgrade itself was breaking